plugins/auth-backend/src/providers/google/provider.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import passport from 'passport';
+import express from 'express';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+import {
+  AuthProvider,
+  AuthProviderRouteHandlers,
+  AuthResponse,
+} from './../types';
+
+const postMessageResponse = (res: express.Response, data: AuthResponse) => {
+  const jsonData = JSON.stringify(data);
+  const base64Data = Buffer.from(jsonData, 'utf8').toString('base64');
+
+  res.setHeader('X-Frame-Options', 'sameorigin');
+  res.end(`
+<html>
+<body>
+  <script>
+    (window.opener || window.parent).postMessage(JSON.parse(atob('${base64Data}')), location.origin)
+  </script>
+</body>
+</html>
+  `);
+};
```

Comment:
is this generic enough to be in a util or common folder, outside the google folder? it's like almost part of the general protocol isn't it

Reply:
Yes it is. Will move it to a util folder. Wanted to do it when we have another provider in.

Reply:
That's fine, just checking